dask/array/tests/test_xarray.py

```diff
@@ -22,3 +22,15 @@ def test_asanyarray():
     y = da.asanyarray(xr.DataArray([1, 2, 3.0]))
     assert isinstance(y, da.Array)
     assert_eq(y, y)
+
+
+def test_asarray_recognises_xarray_data():
```

Comment:
It'd be good to have the name of this test indicate it's testing a specific workaround

```suggestion
def test_asarray_xarray_intersphinx_workaround():
```